### PR TITLE
30127 - TeachingPopover updates

### LINKED ISSUE #30127: Teaching Popover PR Feedback
This issue is tracking the resolution of outstanding comments from PR [feat(react-teaching-popover-preview): Core functionality](https://github.com/microsoft/fluentui/pull/29894)

Unresolved comments from @layershifter:
- [ ] [Use images from CDN in examples](https://github.com/microsoft/fluentui/pull/29894#discussion_r1427770397) 
- [ ] [Fix use of `renderByDefault` in `useTeachingPopoverHeader`](https://github.com/microsoft/fluentui/pull/29894#discussion_r1427795245)
- [ ] [Consider changing `TeachingPopoverButton.buttonType` prop](https://github.com/microsoft/fluentui/pull/29894#discussion_r1427788954) 
- [ ] [Consider changing `TeachingPopoverButton.altStepText` prop](https://github.com/microsoft/fluentui/pull/29894#discussion_r1427789461) 
- [ ] [Remove `Paritial<>` around slots type for `TeachingPopoverPageCountProps`](https://github.com/microsoft/fluentui/pull/29894#discussion_r1427797604) 
- [ ] [Don't use render props: `TeachingPopoverPageCountChildRenderFunction`](https://github.com/microsoft/fluentui/pull/29894#discussion_r1427798617) 
- [ ] [Rename useTeachingPopoverContext.ts to useTeachingPopoverContextValues.ts](https://github.com/microsoft/fluentui/pull/29894#discussion_r1430670341)
- [ ] [Add `defaultCurrentPage` prop to `TeachingPopover`](https://github.com/microsoft/fluentui/pull/29894#discussion_r1430670838) 
- [ ] Update `useTeachingPopoverContextValues` to match what's in `renderTeachingPopover`, and [use `contextValues.teachingPopover` in the render function](https://github.com/microsoft/fluentui/pull/29894#discussion_r1430671019) 
- [ ] [Remove `TeachingPopoverTitle.showDismiss` prop because it's redundant with `dismissButton={null}`](https://github.com/microsoft/fluentui/pull/29894#discussion_r1430675393) 

Unresolved comments from @behowell:
- [ ] [`renderTeachingPopoverPageCount` shouldn't return null](https://github.com/microsoft/fluentui/pull/29894#discussion_r1428373363)
- [ ] [TeachingPopoverPageCount shouldn't render its `carouselIcon` slot multiple times](https://github.com/microsoft/fluentui/pull/29894#discussion_r1432001393)

### PR DESCRIPTION
**Pull request by Mitch-At-Work**
## New Behavior
- Carousel functionality has been unified into TeachingPopoverCarousel
- Moved teaching popover page context to within carousel
- Render page count nav icons as individual components (not slots)
- New 'TeachingPopoverFooter' to replace TeachingPopverActions

## Related Issue(s)

- Fixes #30127

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                                   |     Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                      Change |
| :---------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: | ----------------------------: | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-accordion</samp> <br /> <abbr title='bundle-size/Accordion.fixture.js'>Accordion (including children components)</abbr> | `93.962 kB`<br />`28.706 kB` |  `95.562 kB`<br />`29.292 kB` |   `1.6 kB` <br />`586 B`  |
| <samp>react-calendar-compat</samp> <br /> <abbr title='bundle-size/Calendar.fixture.js'>Calendar Compat</abbr>                      | `142.28 kB`<br />`37.196 kB` | `143.876 kB`<br />`37.732 kB` | `1.596 kB` <br />`536 B`  |
| <samp>react-popover</samp> <br /> <abbr title='bundle-size/Popover.fixture.js'>Popover</abbr>                                       | `121.32 kB`<br />`38.288 kB` | `122.243 kB`<br />`38.599 kB` |    `923 B` <br />`311 B`  |
| <samp>react-portal</samp> <br /> <abbr title='bundle-size/Portal.fixture.js'>Portal</abbr>                                          |   `12.788 kB`<br />`4.67 kB` |   `12.836 kB`<br />`4.705 kB` |      `48 B` <br />`35 B`  |
| <samp>react-provider</samp> <br /> <abbr title='bundle-size/FluentProvider.fixture.js'>FluentProvider</abbr>                        |  `22.928 kB`<br />`8.472 kB` |   `22.962 kB`<br />`8.498 kB` |      `34 B` <br />`26 B`  |
| <samp>react-toast</samp> <br /> <abbr title='bundle-size/Toast.fixture.js'>Toast (including Toaster)</abbr>                         | `93.274 kB`<br />`28.044 kB` |  `95.495 kB`<br />`28.639 kB` | `2.221 kB` <br />`595 B`  |
| <samp>react-tooltip</samp> <br /> <abbr title='bundle-size/Tooltip.fixture.js'>Tooltip</abbr>                                       | `53.689 kB`<br />`18.964 kB` |  `53.732 kB`<br />`18.988 kB` |      `43 B` <br />`24 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                          |         Size (minified/GZIP) |
| -------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>global-context</samp> <br /> <abbr title='bundle-size/createContext.fixture.js'>createContext</abbr>                 |         `510 B`<br />`328 B` |
| <samp>global-context</samp> <br /> <abbr title='bundle-size/createContextSelector.fixture.js'>createContextSelector</abbr> |         `537 B`<br />`339 B` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/Badge.fixture.js'>Badge</abbr>                                    |  `26.905 kB`<br />`8.729 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/CounterBadge.fixture.js'>CounterBadge</abbr>                      |  `27.806 kB`<br />`9.024 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/PresenseBadge.fixture.js'>PresenceBadge</abbr>                    |  `25.311 kB`<br />`9.306 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                                 | `39.513 kB`<br />`11.169 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr>                 | `46.874 kB`<br />`12.661 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>                         | `44.292 kB`<br />`12.542 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr>                       | `52.306 kB`<br />`14.134 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>                     | `56.559 kB`<br />`13.068 kB` |
| <samp>react-divider</samp> <br /> <abbr title='bundle-size/Divider.fixture.js'>Divider</abbr>                              |   `20.82 kB`<br />`7.763 kB` |
| <samp>react-image</samp> <br /> <abbr title='bundle-size/Image.fixture.js'>Image</abbr>                                    |  `15.736 kB`<br />`6.228 kB` |
| <samp>react-jsx-runtime</samp> <br /> <abbr title='bundle-size/Classic.fixture.js'>Classic Pragma</abbr>                   |      `1.057 kB`<br />`530 B` |
| <samp>react-jsx-runtime</samp> <br /> <abbr title='bundle-size/DevRuntime.fixture.js'>JSX Dev Runtime</abbr>               |   `3.783 kB`<br />`1.648 kB` |
| <samp>react-jsx-runtime</samp> <br /> <abbr title='bundle-size/Runtime.fixture.js'>JSX Runtime</abbr>                      |   `4.377 kB`<br />`1.881 kB` |
| <samp>react-label</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Label</abbr>                                    |  `14.266 kB`<br />`5.828 kB` |
| <samp>react-link</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Link</abbr>                                      |  `17.082 kB`<br />`6.911 kB` |
| <samp>react-overflow</samp> <br /> <abbr title='bundle-size/hooks.fixture.js'>hooks only</abbr>                            |  `12.862 kB`<br />`4.827 kB` |
| <samp>react-positioning</samp> <br /> <abbr title='bundle-size/usePopper.fixture.js'>usePositioning</abbr>                 |  `26.336 kB`<br />`9.515 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Default.fixture.js'>Text - Default</abbr>                     |  `16.705 kB`<br />`6.569 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Wrappers.fixture.js'>Text - Wrappers</abbr>                   |  `19.878 kB`<br />`6.896 kB` |
| <samp>react-utilities</samp> <br /> <abbr title='bundle-size/SSRProvider.fixture.js'>SSRProvider</abbr>                    |         `180 B`<br />`160 B` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/491d0cb952b14057711ff36348da50db92eaf78d'>491d0cb952b14057711ff36348da50db92eaf78d</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">69</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">64</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/Avatar-mount_base.html">649</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/Avatar-mount.html">657</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/Button-mount_base.html">295</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/Button-mount.html">303</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/Field-mount_base.html">1184</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/Field-mount.html">1109</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/FluentProvider-mount_base.html">679</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/FluentProvider-mount.html">700</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/FluentProviderWithTheme-mount_base.html">75</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/FluentProviderWithTheme-mount.html">84</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">62</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">60</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">69</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">64</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/MakeStyles-mount_base.html">843</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/MakeStyles-mount.html">853</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/Persona-mount_base.html">1754</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/Persona-mount.html">1738</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/SpinButton-mount_base.html">1414</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30423/perf-test-react-components/SpinButton-mount.html">1387</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: change/@fluentui-react-popover-97a78330-2be0-41c5-a460-4a1d2f2106d5.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-popover-97a78330-2be0-41c5-a460-4a1d2f2106d5.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=30423&commitId=17cb94540a592b6cad3fb3511d142b5103949eec&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.basic - RTL.normal.chromium.png|0| Added|
|Avatar Converged.basic.normal.chromium.png|0| Added|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-popover/src/popoverContext.ts [hunk 2]
```diff
@@ -45,3 +45,43 @@ export type PopoverContextValue = Pick<
 
 export const usePopoverContext_unstable = <T>(selector: ContextSelector<PopoverContextValue, T>): T =>
   useContextSelector(PopoverContext, (ctx = popoverContextDefaultValue) => selector(ctx));
+
+export function usePopoverContextValues_unstable(state: PopoverState): PopoverContextValue {
```
Comment (layershifter) [inline packages/react-components/react-popover/src/popoverContext.ts:49]: Let's move this change into a separate PR.
Also, I don't see this hook to be used anywhere in `Popover` code.

File: packages/react-components/react-teaching-popover-preview/src/components/TeachingPopover/TeachingPopover.types.ts [hunk 3]
```diff
@@ -1,60 +1,11 @@
...
-  /**
-   * The appearance property (extended from popover, but removed 'inverted').
-   */
-  appearance?: TeachingPopoverAppearance;
-};
-
-export type TeachingPopoverContextValues = {
-  teachingPopover: TeachingPopoverContextValue;
-};
+export type TeachingPopoverProps = PopoverProps & {};
```
Comment (layershifter) [inline packages/react-components/react-teaching-popover-preview/src/components/TeachingPopover/TeachingPopover.types.ts:6]:
```suggestion
export type TeachingPopoverProps = PopoverProps;
```

File: packages/react-components/react-teaching-popover-preview/src/components/TeachingPopover/renderTeachingPopover.tsx [hunk 4]
```diff
@@ -1,58 +1,16 @@
...
-    setOpen,
-    size,
-    toggleOpen,
-    trapFocus,
-    triggerRef,
-    withArrow,
-    inertTrapFocus,
-  } = state;
-
+export const renderTeachingPopover_unstable = (state: TeachingPopoverState, contextValues: PopoverContextValue) => {
```
Comment (layershifter) [inline packages/react-components/react-teaching-popover-preview/src/components/TeachingPopover/renderTeachingPopover.tsx:8]: Out of curiosity, can't we do `export const renderTeachingPopover_unstable = renderPopover_unstable`?

File: packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/TeachingPopoverCarousel.tsx [hunk 5]
```diff
@@ -18,7 +19,9 @@ export const TeachingPopoverCarousel: ForwardRefComponent<TeachingPopoverCarouse
 
     useCustomStyleHook_unstable('useTeachingPopoverCarouselStyles_unstable')(state);
 
-    return renderTeachingPopoverCarousel_unstable(state);
+    const context = useTeachingPopoverCarouselContextValues_unstable(state);
+
+    return renderTeachingPopoverCarousel_unstable(state, context);
```
Comment (layershifter) [inline packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/TeachingPopoverCarousel.tsx:24]:
```suggestion
    const contextValues = useTeachingPopoverCarouselContextValues_unstable(state);

    return renderTeachingPopoverCarousel_unstable(state, contextValues);
```

File: packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/TeachingPopoverCarouselContext.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  const { currentPage, setCurrentPage, totalPages, onPageChange } = state;
+
+  const teachingPopoverCarousel = {
+    currentPage,
+    setCurrentPage,
+    totalPages,
+    onPageChange,
+  };
+
+  return teachingPopoverCarousel;
```
Comment (layershifter) [inline packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/TeachingPopoverCarouselContext.ts:45]:
```suggestion
  const carousel = {
    currentPage,
    setCurrentPage,
    totalPages,
    onPageChange,
  };

  return carousel;
```

As there could be multiple context providers, we return an object that could be applied to multiple providers i.e. `<ContextA value={contextValues.a} />`

File: packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/TeachingPopoverCarousel.types.ts [hunk 7]
```diff
@@ -1,20 +1,121 @@
...
 /**
  * TeachingPopoverCarousel Props
  */
-export type TeachingPopoverCarouselProps = ComponentProps<TeachingPopoverCarouselSlots>;
+export type TeachingPopoverCarouselProps = Partial<ComponentProps<TeachingPopoverCarouselSlots>> & {
+  /**
+   * Controls whether buttons will be centered (balanced) or right aligned
+   * Defaults to 'centered'.
+   */
+  carouselLayout?: TeachingPopoverCarouselLayout;
```
Comment (layershifter) [inline packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/TeachingPopoverCarousel.types.ts:70]:
```suggestion
  layout?: TeachingPopoverCarouselLayout;
```

Just `layout`?

File: packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/TeachingPopoverCarousel.types.ts [hunk 8]
```diff
@@ -1,20 +1,121 @@
...
+export type TeachingPopoverStrings = {
+  previous: string;
+  initialStepText: string;
+  next: string;
+  finalStepText: string;
+  /**
+   * Localize the separator between numbers, or use a function to fully override
+   */
+  pageCountText?: TeachingPopoverPageCountRenderType;
+};
```
Comment (layershifter) [inline packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/TeachingPopoverCarousel.types.ts:53]: FYI @ling1726

File: packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/TeachingPopoverCarousel.types.ts [hunk 9]
```diff
@@ -1,20 +1,121 @@
...
+
+  /**
+   * Callback to notify next page was clicked
+   */
+  onClickNext?: EventHandler<TeachingPopoverPageChangeData>;
+
+  /**
+   * Callback to notify next page was clicked
+   */
+  onClickPrevious?: EventHandler<TeachingPopoverPageChangeData>;
```
Comment (layershifter) [inline packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/TeachingPopoverCarousel.types.ts:101]: Curious why these are needed, can't I do just `<TeachingPopoverCarouselProps next={{ onClick }} />`?

File: packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/useTeachingPopoverCarouselStyles.styles.ts [hunk 10]
```diff
@@ -1,21 +1,109 @@
...
+  state.footer.className = mergeClasses(
+    teachingPopoverCarouselClassNames.footer,
+    styles.footer,
+    carouselLayout === 'centered' ? styles.footerCentered : styles.footerRightAligned,
+    state.footer.className,
+  );
+  if (state.previous) {
+    state.previous.className = mergeClasses(
+      teachingPopoverCarouselClassNames.previous,
+      appearance === 'brand' ? styles.brandPrevious : styles.previous,
```
Comment (layershifter) [inline packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/useTeachingPopoverCarouselStyles.styles.ts:88]:
```suggestion
      styles.previous,
      appearance === 'brand' && styles.brandPrevious,
```

Let's avoid styles duplication and apply base styles always and brand styles on top of it

File: packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarouselNav/useTeachingPopoverCarouselNav.tsx [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+  ref: React.Ref<HTMLDivElement>,
+): TeachingPopoverCarouselNavState => {
+  const focusableGroupAttr = useFocusableGroup({ tabBehavior: 'limited' });
+  const totalPages = useTeachingPopoverCarouselContext_unstable(context => context.totalPages);
+  const currentPage = useTeachingPopoverCarouselContext_unstable(context => context.currentPage);
+
+  // Generate the child RatingItems and memoize them to prevent unnecessary re-rendering
+  const rootChildren = React.useMemo(() => {
+    return Array.from(Array(totalPages), (_, i) => (
+      <TeachingPopoverCarouselNavIcon key={'TeachingPopoverCarouselNavIcon-' + i} index={i} />
```
Comment (layershifter) [inline packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarouselNav/useTeachingPopoverCarouselNav.tsx:28]:
```suggestion
      <TeachingPopoverCarouselNavIcon key={i} index={i} />
```

Keys are scoped, so there is no need to complicate them 😉

File: packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarouselNav/useTeachingPopoverCarouselNav.tsx [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+    components: {
+      root: 'div',
+    },
+    root: slot.always(
+      getIntrinsicElementProps('div', {
+        ref,
+        role: 'list',
+        tabIndex: 0,
+        children: rootChildren,
+        ...props,
```
Comment (layershifter) [inline packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarouselNav/useTeachingPopoverCarouselNav.tsx:44]: Is this intentional? If user will pass own children, defined in component will be gone

File: packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarouselNav/useTeachingPopoverCarouselNav.tsx [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+ */
+export const useTeachingPopoverCarouselNav_unstable = (
+  props: TeachingPopoverCarouselNavProps,
+  ref: React.Ref<HTMLDivElement>,
+): TeachingPopoverCarouselNavState => {
+  const focusableGroupAttr = useFocusableGroup({ tabBehavior: 'limited' });
+  const totalPages = useTeachingPopoverCarouselContext_unstable(context => context.totalPages);
+  const currentPage = useTeachingPopoverCarouselContext_unstable(context => context.currentPage);
+
+  // Generate the child RatingItems and memoize them to prevent unnecessary re-rendering
```
Comment (layershifter) [inline packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarouselNav/useTeachingPopoverCarouselNav.tsx:25]:
```suggestion
  // Generate the child TeachingPopoverCarouselNavIcon and memoize them to prevent unnecessary re-rendering
```

Copy-paste issue?

File: packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarouselNavIcon/TeachingPopoverCarouselNavIcon.types.ts [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+export type TeachingPopoverCarouselNavIconSlots = {
+  /**
+   * The root slot of the TeachingPopoverCarouselNavIcon.
+   * Default html element is div
+   */
+  root: NonNullable<Slot<'div'>>;
+  /**
+   * Nav icon set as an ARIA compliant button
+   */
+  navButton: NonNullable<Slot<'button'>>;
```
Comment (layershifter) [inline packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarouselNavIcon/TeachingPopoverCarouselNavIcon.types.ts:13]: I have to say that naming is weird there. IMO it should be `TeachingPopoverCarouselNavButton` as icons are non-interactive items

File: packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverFooter/useTeachingPopoverFooter.ts [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+    },
+    renderByDefault: true,
+    elementType: Button,
+  });
+
+  // Merge any provided callback with close trigger
+  const secondaryButtonClick = mergeCallbacks(handleButtonClick, secondary?.onClick);
+  if (secondary) {
+    secondary.onClick = secondaryButtonClick;
+  }
```
Comment (layershifter) [inline packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverFooter/useTeachingPopoverFooter.ts:46]:
```suggestion
  if (secondary) {
    secondary.onClick = mergeCallbacks(handleButtonClick, secondary.onClick);
  }
```

?

File: packages/react-components/react-teaching-popover-preview/src/components/TeachingPopover/renderTeachingPopover.tsx (hunk 4 above)
Comment (Mitch-At-Work) [reply packages/react-components/react-teaching-popover-preview/src/components/TeachingPopover/renderTeachingPopover.tsx:8]: This was mostly an attempt to move to the above format where context is passed in from the wrapper class, I'll set this to be equal to renderPopover for now, but would probably be good to update Popover and this as a seperate PR

File: packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/TeachingPopoverCarousel.types.ts (hunk 9 above)
Comment (Mitch-At-Work) [reply packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/TeachingPopoverCarousel.types.ts:101]: `Good catch, merging these instead`

File: packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarouselNav/useTeachingPopoverCarouselNav.tsx (hunk 12 above)
Comment (Mitch-At-Work) [reply packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarouselNav/useTeachingPopoverCarouselNav.tsx:44]: I figured I'd give users the option but fair point they can probably just replace this component if desired, will move to ensure users don't get confused by accident.

File: packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarouselNavIcon/__snapshots__/TeachingPopoverCarouselNavIcon.test.tsx.snap [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+exports[`TeachingPopoverCarouselNavIcon renders a default state 1`] = `
+<div>
+  <div
+    class="fui-TeachingPopoverCarouselNavIcon"
+  >
+    <button
+      class="fui-TeachingPopoverCarouselNavIcon__navButton"
+      type="button"
+    />
+  </div>
```
[the remark this answers is not in the document]
Comment (Mitch-At-Work) [reply packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarouselNavIcon/__snapshots__/TeachingPopoverCarouselNavIcon.test.tsx.snap:12]: The underlying button relies on opacity for styling, but we want no opacity on the border/focus styles.

Usually, I would just achieve this via background opacity value, but since we use tokens it would be a bit weird string logic to ensure, so just used css opacity on internal button with wrapper as focus/border.

**Review by layershifter — APPROVED** [review]

**Review by mltejera — APPROVED** [review]
